components/datetime/src/datetime.rs

```diff
@@ -21,7 +25,623 @@ use icu_locid::{extensions_unicode_key as key, Locale};
 use icu_plurals::provider::OrdinalV1Marker;
 use icu_provider::prelude::*;
 
-use crate::{date::DateTimeInput, CldrCalendar, DateTimeFormatError, FormattedDateTime};
+use crate::{date::DateTimeInput, CldrCalendar, DateTimeFormatError, FormattedDateTime};
+
+/// [`TimeFormat`] is a structure of the [`icu_datetime`] component that provides time formatting only.
+/// When constructed, it uses data from the [data provider], selected [`Locale`] and provided preferences to
+/// collect all data necessary to format any time into that locale.
+///
+/// For that reason, one should think of the process of formatting a time in two steps - first, a computational
+/// heavy construction of [`TimeFormat`], and then fast formatting of [`DateTimeInput`] data using the instance.
+///
+/// [`icu_datetime`]: crate
+/// [`DateTimeFormat`]: crate::datetime::TimeFormat
+///
+/// # Examples
+///
+/// ```
+/// use icu::calendar::{DateTime, Gregorian};
+/// use icu::datetime::TimeFormat;
+/// use icu::locid::locale;
+/// use icu_provider::inv::InvariantDataProvider;
+///
+/// let provider = InvariantDataProvider;
+///
+/// let tf = TimeFormat::<Gregorian>::try_new_short(locale!("en"), &provider, None)
+///     .expect("Failed to create DateTimeFormat instance.");
+///
+/// let datetime = DateTime::new_gregorian_datetime(2020, 9, 1, 12, 34, 28)
+///     .expect("Failed to construct DateTime.");
+///
+/// let value = tf.format_to_string(&datetime);
+/// ```
+///
+/// This model replicates that of `ICU` and `ECMA402`.
+///
+/// [data provider]: icu_provider
+pub struct TimeFormat<C>(pub(super) raw::TimeFormat, PhantomData<C>);
```

Comment:
Note: The calendar `C` is for date calendars. Don't include it on TimeFormat.

Reply:
I'm going to defer this to a follow up, as this will require changes to the CLDR data import code to strip out calendar information when working with time patterns and symbols. We'll also have to add new fallbacking code so that we drop the calendar information when accessing time data from inside `DateTimeFormat`. I think this is too much to do as part of this PR, it's already quite large.

Reply:
ok; please make sure this action item is tracked somewhere.

Reply:
Filed https://github.com/unicode-org/icu4x/issues/2153 for this.